spec/consensus/proposer-based-timestamp/pbts_001_draft.md

```diff
@@ -0,0 +1,268 @@
+# Proposer-Based Time
+
+## Current BFTTime 
+
+### Description
+
+In Tendermint consensus, the first version of how time is computed and stored in a block works as follows:
+- validators send their current local time as part of `precommit` messages
+- upon collecting the `precommit` messages that the proposer uses to build a commit to be put in the next block, the proposer computes the `time` of the next block as the median (weighted over voting power) of the times in the `precommit` messages.
+
+### Analysis
+
+1. **Fault tolerance.** The computed median time is called [`bfttime`][bfttime] as it is indeed fault-tolerant: if **less than a third** of the validators is faulty (counted in voting power), it is guaranteed that the computed time lies between the minimum and the maximum times sent by correct validators.
+2. **Effect of faulty validators.** If more than `1/2` of the voting power (which is in fact more than one third and less than two thirds of the voting power) is held by faulty validators, then the time is under total control of the faulty validators. (This is particularly challenging in the context of [lightclient][lcspec] security.)
+3. **Proposer influence on block time.** The proposer of the next block has a degree of freedom in choosing the `bfttime`, since it computes the median time based on the timestamps from `precommit` messages sent by 
+`2f + 1` correct validators.
+   1. If there are `n` different timestamps in the  `precommit` messages, the proposer can use any subset of timestamps that add up to `2f + 1` 
+   of the voting power in order to compute the median.
+   1. If the validators decide in different rounds, the proposer can decide on which round the median computation is based. 
```

Comment:
Yes, if +2/3 precommit in round 1 but timeout before they receive these precommits (that is, before they decide) they go to the next round. In round 2 they might precommit again. By agreement it will be the same block, but the times in their precommits will be different.

So you have two rounds with +2/3 precommits. Both are valid commit for the block. The proposer can pick which one it prefers.

Reply:
Yup thanks